src/service/product-service.ts

```diff
@@ -457,15 +457,26 @@ export default class ProductService {
     for (let i = 0; i < containers.length; i += 1) {
       const c = containers[i];
       // eslint-disable-next-line no-await-in-loop
-      await ContainerRevision.findOne({ where: { container: { id: c.id }, revision: c.revision }, relations: ['products', 'products.product'] }).then(async (revision) => {
-        const update: UpdateContainerParams = {
-          products: revision.products.map((p) => p.product.id),
-          public: c.public,
-          name: revision.name,
-          id: c.id,
-        };
-        await ContainerService.directContainerUpdate(update);
-      });
+      const revision = await ContainerRevision.findOne({ where: { container: { id: c.id }, revision: c.revision }, relations: ['products', 'products.product'] });
+      const update: UpdateContainerParams = {
+        products: revision.products.filter((p) => p.product && p.product.deletedAt == null).map((p) => p.product.id),
+        public: c.public,
+        name: revision.name,
+        id: c.id,
+      };
+      await ContainerService.directContainerUpdate(update);
     }
   }
+
+  /**
+   * (Soft) delete a product
+   * @param productId
+   */
+  public static async deleteProduct(productId: number): Promise<void> {
```

Comment:
Doesn't it make more sense that this triggers a update propegation? If I delete a product it should also be removed out of all containers and POS right?

Reply:
I also thought about it some more yesterday and today. By definition of the soft deletions, it should not be necessary (as deleted products should never, ever be returned again). It also reduces the amounts of "states" you can have, because if you always propagate after a soft delete, what happens if - somehow - you don't?

I'll keep these tests in place to verify that deleted products/containers/POSs are never returned again, but I will add the propagation. This is more to meet possible future use cases, like the one you mentioned over Whatsapp: changing transactions and wanting to know what products where in that specific container. It is possible with the current implementation, but it is slightly more complex.